fastlane_bot/bot.py

```diff
@@ -264,46 +243,27 @@ def _get_deadline(self, block_number) -> int:
         int
             The deadline (as UNIX epoch).
         """
-        block_number = (
-            self.ConfigObj.w3.eth.block_number if block_number is None else block_number
-        )
-        return (
-            self.ConfigObj.w3.eth.get_block(block_number).timestamp
-            + self.ConfigObj.DEFAULT_BLOCKTIME_DEVIATION
-        )
-
-    @classmethod
-    def _get_arb_finder(cls, arb_mode: str) -> Callable:
-        return cls.ARB_FINDER[arb_mode]
+        if block_number is None:
+            block_number = self.ConfigObj.w3.eth.block_number
+        return self.ConfigObj.w3.eth.get_block(block_number).timestamp + self.ConfigObj.DEFAULT_BLOCKTIME_DEVIATION
 
-    def _find_arbitrage(
-        self,
-        flashloan_tokens: List[str],
-        CCm: CPCContainer,
-        arb_mode: str,
-        randomizer: int
-    ) -> dict:
-        arb_finder = self._get_arb_finder(arb_mode)
-        random_mode = arb_finder.AO_CANDIDATES if randomizer else None
-        finder = arb_finder(
-            flashloan_tokens=flashloan_tokens,
-            CCm=CCm,
-            mode="bothin",
-            result=random_mode,
-            ConfigObj=self.ConfigObj,
-        )
-        return {"finder": finder, "r": finder.find_arbitrage()}
+    def get_arb_finder(self, arb_mode: str, flashloan_tokens, CCm) -> ArbitrageFinderBase:
+        return {
```

Comment:
not a biggie but I personally would define the dict in a class constant that can be inspected when looking at the class and then reference it here rather than defining it only inline.

This also has the advantage that you can include it in tests, eg you can check that the dict keys of available methods are what you expected they are [removing a key there is probably a semver major version event]